packages/frontend/core/src/pages/workspace/detail-page/detail-page.tsx

```diff
@@ -94,9 +94,22 @@ const DetailPageImpl = memo(function DetailPageImpl() {
   }, [editor, isActiveView, setActiveBlockSuiteEditor]);
 
   useEffect(() => {
-    AIProvider.slots.requestOpenWithChat.on(() => {
-      rightSidebar.open();
-      if (activeTabName !== 'chat') {
+    AIProvider.slots.requestOpenWithChat.on(params => {
+      const opened = rightSidebar.isOpen$.value;
+      const actived = activeTabName === 'chat';
+
+      if (!opened) {
+        rightSidebar.open();
+      }
+
+      // Save chat parameters:
+      // * The right sidebar is not open
+      // * Chat panel is not activated
+      if (!opened || !actived) {
+        AIProvider.sharedChatParams = params;
```

Comment:
I think you can directly modify the chat params of the chat panel component directly here without introducing sharedChatParams in AIProvider. Having a global state to be produced/consumed seems to be too frigile.

E.g., 

```ts
// add AIProvider.requestChatPanelElement api to get the chat panel instance of the given editor
// make it promise to make sure it opens and ready to be used
// type requestChatPanelElement = (host: EditorHost) => Promise<ChatPanel | null>;
AIProvider.requestChatPanelElement(host).chatParams = params;
```

Reply:
There is a problem here: `ChatPanel` probably hasn't been initialized yet.

Reply:
Yeah, that's why the getter function is an async function. Could add an option to open it.